core/math/color.cpp

```diff
@@ -109,36 +109,29 @@ uint64_t Color::to_rgba64() const {
 	return c;
 }
 
-String _to_hex(float p_val) {
+void _append_hex(float p_val, char32_t *string) {
 	int v = Math::round(p_val * 255.0f);
 	v = CLAMP(v, 0, 255);
-	String ret;
 
 	for (int i = 0; i < 2; i++) {
-		char32_t c[2] = { 0, 0 };
-		int lv = v & 0xF;
-		if (lv < 10) {
-			c[0] = '0' + lv;
-		} else {
-			c[0] = 'a' + lv - 10;
-		}
-
+		string[i] = hex_char_table_lower[v & 0xF];
```

Comment:
I would have removed the loop, as the compiler will do it anyway, but it will produce the same code. (it's just more code like this)

Reply:
If it doesn't make a performance different I try to choose the code that is easier to read. Which would be the loop code, for me anyway :)

Reply:
Yeah, I don't see any compiler not unrolling a 2 iterations loop, so it works for me !